protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,341 @@
+        treasury = _treasury;
+        auctionStarter = _auctionStarter;
+        emit AuctionStarterSet(_auctionStarter);
+    }
+
+    /**
+     * @notice Set auction configuration
+     * @param _config Auction configuration
+     */
+    function setAuctionConfig(AuctionConfig calldata _config) external override onlyElevatedAccess {
+        if (_config.auctionStartCooldown == 0
+                || _config.auctionMinimumDistributedGold == 0
+                || _config.auctionsTimeDiff == 0) 
+            { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        uint256 currentEpochCache = _currentEpochId;
+        if (!epochs[currentEpochCache].hasEnded()) { revert InvalidOperation(); }
+        auctionConfig = _config;
+
+        emit AuctionConfigSet(currentEpochCache, _config);
+    }
+
+    /**
+     * @notice Set address to trigger auction start. Zero address accepted
+     * @param _starter Auction starter
+     */
+    function setAuctionStarter(address _starter) external override onlyElevatedAccess {
+        /// @notice No zero address checks. Zero address is a valid input
+        auctionStarter = _starter;
+        if (!epochs[_currentEpochId].hasEnded()) { revert InvalidOperation(); }
+        emit AuctionStarterSet(_starter);
+    }
+
+    /**
+     * @notice Set token used for bidding
+     * @param _bidToken Bid token
+     */
+    function setBidToken(address _bidToken) external override onlyElevatedAccess {
+        if (_bidToken == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (!epochs[_currentEpochId].hasEnded()) { revert InvalidOperation(); }
+        bidToken = IERC20(_bidToken);
+        emit BidTokenSet(_bidToken);
+    }
+
+    /**
+     * @notice Start auction. Auction start can be triggered by anyone if `auctionStarter` not set
+     * @dev The Temple Gold amount for the auction is fixed and set at startAuction(). 
+     * So in `startAuction()`, there is a call to `_distributeGold()` to mint and distribute TGOLD. 
+     * Any other `_distributeGold()` calls during auction is tracked for next auction use.
+     */
+    function startAuction() external override {
+        if (auctionStarter != address(0) && msg.sender != auctionStarter) { revert CommonEventsAndErrors.InvalidAccess(); }
+        uint256 currentEpochCache = _currentEpochId;
+        EpochInfo storage prevAuctionInfo = epochs[currentEpochCache];
+        if (!prevAuctionInfo.hasEnded()) { revert CannotStartAuction(); }
+       
+        AuctionConfig storage config = auctionConfig;
+        /// @notice last auction end time plus wait period
+        if (currentEpochCache > 0 && (prevAuctionInfo.endTime + config.auctionsTimeDiff > block.timestamp)) {
+            revert CannotStartAuction();
+        }
+        _distributeGold();
+        uint256 totalGoldAmount = nextAuctionGoldAmount;
+        nextAuctionGoldAmount = 0;
+        uint256 epochId = _currentEpochId = currentEpochCache + 1;
```

Comment:
not important but interesting:

everytime `recoverToken()` is called, that epoch becomes a ghost auction with everything set at 0. Not important, but good to know for frontend stuff. 

This happens because `recoverToken()` only deletes `epoch[i]` but doesn't change `currentEpochId`, and then when `startAuction()` is called, then the `epochId` of the new auction is `i+1`. So auction `i` is left as a ghost auction.

Reply:
Thanks for pointing out